mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBNativeQueryPaginationTest.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.test.query.nativequery;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.ogm.OgmSession;
+import org.hibernate.ogm.utils.OgmTestCase;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * The test pagination behaviour for native queries.
+ * Using find and aggregate pipeline.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class MongoDBNativeQueryPaginationTest extends OgmTestCase {
```

Comment:
I think it would be nice to have a similar test in core as well for HQL queries.
Maybe you can add it reusing this code? Not sure.

Reply:
Could be nice! Maybe we could create another Issue for that?

Reply:
We could, but it's sort of related to what you are doing and I don't feel the need to create a separate issue for it. But I don't have a strong opinion on this, feel free to d oas you prefer.